reana_client/utils.py

```diff
@@ -173,6 +180,49 @@ def _prepare_kwargs(reana_yaml):
         raise e
 
 
+def _validate_environment_tags(reana_yaml):
+    """Validate environments in REANA specification file according to workflow type.
+
+    :param reana_yaml: Dictionary which represents REANA specifications file.
+    :return: True if environment tags are valid.
+    """
+    workflow_type = reana_yaml["workflow"]["type"]
+
+    if workflow_type == "serial":
+        workflow_steps = reana_yaml["workflow"]["specification"]["steps"]
+        _validate_serial_workflow_environment(workflow_steps)
+    elif workflow_type == "yadage":
+        logging.info("Not Implemented yet.")
+    elif workflow_type == "cwl":
+        logging.info("Not Implemented yet.")
+
+    return True
+
+
+def _validate_serial_workflow_environment(workflow_steps):
+    """Validate environments in REANA serial workflow.
+
+    :param workflow_steps: List of dictionary which represents different steps involved in workflow.
+    :raises Warning: Warns user if valid tags are not used in serial workflow steps.
+    """
+    for step in workflow_steps:
+        environment = step["environment"].split(":")
+        environment_tag = environment[-1]
+        if len(environment) == 1:
+            logging.warning(
+                "Include a tag in the enviornment {}".format(environment[0])
+            )
+            sys.exit(1)
+        elif len(environment) == 2 and environment_tag in WARNING_TAGS:
```

Comment:
I think that if you pass something like `foo:bar:baz` it returns `True` as its length is not 1 nor 2 (untested).

Reply:
I can do like this:

```
if len()>2 or len()==0
  Warning: Invalid environment format. Use x:y
elif len() == 1:
  Warning: Include a tag
elif len() == 2 and environment_tag in WARNING_TAGS:
  Warning: Invalid tag used
```

wdyt?

Reply:
The format without `:`, for example `myorg/myimage`, is perfectly valid, so please beware about wording.

Regarding implementation for `foo:bar:baz:fuux:quux`, you can do upfront something like `split(":", 1)` to have only two elements, the image name and the image tag, and then check whether the detected tag contains more `:` inside. That might yield simpler code logic.